include/drivers/interrupt_controller/gic.h

```diff
@@ -219,6 +210,115 @@
 
+/* GICC_IAR */
+#define GICC_IAR_SPURIOUS		1023
+
+/* GICD_ICFGR */
+#define GICD_ICFGR_MASK			BIT_MASK(2)
+#define GICD_ICFGR_TYPE			BIT(1)
+
+/* GICD_TYPER.ITLinesNumber 0:4 */
+#define IT_LINES_NO_MASK		0x1f
+
+/*
+ * Helper Constants
+ */
+
+#define	GIC_SPI_INT_BASE		32
+
+#define NUM_INTR_PER_REG		32
+
+#define NUM_CFG_PER_REG			16
+
+#define NUM_PRI_PER_REG			4
+
+/* GIC idle priority : value '0xff' will allow all interrupts */
+#define GIC_IDLE_PRIO			0xff
+/* Priority levels 0:255 */
+#define GIC_PRI_MASK			0xff
+
+/*
+ * '0xa0'is used to initialize each interrtupt default priority.
+ * This is an arbitrary value in current context.
+ * Any value '0x80' to '0xff' will work for both NS and S state.
+ * The values of individual interrupt and default has to be chosen
+ * carefully if PMR and BPR based nesting and preemption has to be done.
+ */
+#define GIC_INT_DEF_PRI_X4		0xa0a0a0a0
+
+#if defined(CONFIG_GIC_V3)
+
+#define GIC_RDIST_BASE		DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1)
+
+/* Fixme: update from platform define or dt */
+#define GIC_NUM_CPU_IF			1
+/* Fixme: get current core id from mpidr */
+#define GET_CPUID			0
+
+#define GIC_GET_RDIST(cpuid)		gic_rdists[cpuid]
+
+/* SGI base is at 64K offset from Redistributor */
+#define GICR_SGI_BASE_OFF		0x10000
+
+/* GICR registers offset from RD_base(n) */
+#define GICR_CTLR			0x0000
+#define	GICR_IIDR			0x0004
+#define	GICR_TYPER			0x0008
+#define GICR_STATUSR			0x0010
+#define GICR_WAKER			0x0014
+
+/* Offsets from GICD base or GICR(n) SGI_base */
+#define	GIC_DIST_IGROUPR		0x0080
+#define	GIC_DIST_ISENABLER		0x0100
+#define	GIC_DIST_ICENABLER		0x0180
+#define	GIC_DIST_ISPENDR		0x0200
+#define	GIC_DIST_ICPENDR		0x0280
+#define	GIC_DIST_ISACTIVER		0x0300
+#define	GIC_DIST_ICACTIVER		0x0380
+#define	GIC_DIST_IPRIORITYR		0x0400
+#define	GIC_DIST_ITARGETSR		0x0800
+#define	GIC_DIST_ICFGR			0x0c00
+#define GIC_DIST_IGROUPMODR		0x0d00
+#define	GIC_DIST_SGIR			0x0f00
+
+/* GICD GICR common access macros */
+#define IGROUPR(base, n)		(base + GIC_DIST_IGROUPR + (n) * 4)
+#define ISENABLER(base, n)		(base + GIC_DIST_ISENABLER + (n) * 4)
+#define ICENABLER(base, n)		(base + GIC_DIST_ICENABLER + (n) * 4)
+#define ISPENDR(base, n)		(base + GIC_DIST_ISPENDR + (n) * 4)
+#define ICPENDR(base, n)		(base + GIC_DIST_ICPENDR + (n) * 4)
+#define IPRIORITYR(base, n)		(base + GIC_DIST_IPRIORITYR + n)
+#define ITARGETSR(base, n)		(base + GIC_DIST_ITARGETSR + (n) * 4)
+#define ICFGR(base, n)			(base + GIC_DIST_ICFGR + (n) * 4)
+#define IGROUPMODR(base, n)		(base + GIC_DIST_IGROUPMODR + (n) * 4)
+
+/*
+ * selects redistributor SGI_base for current core for PPI and SGI
+ * selects distributor base for SPI
+ */
+#define GIC_GET_BASE(intid)	((intid < GIC_SPI_INT_BASE) ? \
+				(GIC_GET_RDIST(GET_CPUID) + GICR_SGI_BASE_OFF) \
+				: GIC_DIST_BASE)
+
+/* Register bit definations */
+
+/* GICD_CTLR Interrupt group definitions */
+#define ENABLE_G0			0
+#define ENABLE_G1NS			1
+#define ENABLE_G1S			2
```

Comment:
same deal for these: something like `GICD_CTLR_ENABLE_G0` and so on.

Reply:
ACK